### PR TITLE
Remove IAM V1 conditionals from the UI and Cypress

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

Remove all IAM V1 conditionals from the UI and completely remove IAM V1 from Cypress.

**Please** double-check anything polling related. There was some UI polling that depended on the IAM version endpoint that I'm not 100% on the purpose behind it.

### :athletic_shoe: How to Build and Test the Change

Spin up UI and click around / do :alltheings: IAM related.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### Passing against master

https://buildkite.com/chef/chef-automate-master-verify-private/builds/9103#_

### CODE REVIEW

**Comment by tylercloke** [comment]
Passing against master! https://buildkite.com/chef/chef-automate-master-verify-private/builds/9103#_

File: components/automate-ui/src/app/modules/team/team-add-users/team-add-users.component.ts [hunk 1]
```diff
@@ -56,26 +53,18 @@ export class TeamAddUsersComponent implements OnInit, OnDestroy {
...
-          teamIdA === teamIdB)
-      ).subscribe(([team, teamId]) => {
-        this.team = team;
-        this.teamId = teamId;
-        this.store.dispatch(new GetTeamUsers({ id: teamId}));
-      });
+    this.store.select(teamFromRoute).pipe(
+      takeUntil(this.isDestroyed),
+      map((team) => [team, team ? team.id : null]),
+      filter(([team, _teamId]) => team != null),
```
Comment (msorens) [inline components/automate-ui/src/app/modules/team/team-add-users/team-add-users.component.ts]: Noticed that by moving the filter up one line can simplify it to this:
```
filter(identity),
```

File: components/automate-ui/src/app/pages/project/list/project-list.component.spec.ts [hunk 2]
```diff
@@ -130,7 +129,6 @@ describe('ProjectListComponent', () => {
     element = fixture.debugElement.nativeElement;
     store = TestBed.get(Store);
 
-    store.dispatch(new GetIamVersionSuccess({ version: { major: 'v2' } }));
     fixture.detectChanges();
   });
```
Comment (msorens) [inline components/automate-ui/src/app/pages/project/list/project-list.component.spec.ts]: A few lines below this change
```
displays project data for v2
```
to 
```
displays project data
```

File: components/automate-ui/src/app/modules/token/token-details/api-token-details.component.spec.ts [hunk 3]
```diff
@@ -133,8 +119,6 @@ describe('ApiTokenDetailsComponent', () => {
 
   it('initializes dropdown with those included on the token checked for v2p1', () => {
```
Comment (msorens) [inline components/automate-ui/src/app/modules/token/token-details/api-token-details.component.spec.ts]:
```suggestion
  it('initializes dropdown with those checked on the token', () => {
```

File: components/automate-ui/src/app/modules/policy/list/policy-list.component.spec.ts [hunk 4]
```diff
@@ -68,14 +67,7 @@ describe('PolicyListComponent', () => {
   });
 
   it('displays policy data for v2', () => {
```
Comment (msorens) [inline components/automate-ui/src/app/modules/policy/list/policy-list.component.spec.ts]:
```suggestion
  it('displays policy data', () => {
```

File: components/automate-ui/src/app/modules/team/team-details/team-details.component.ts [hunk 5]
```diff
@@ -141,37 +131,21 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
...
-      filter(([_isV2, team]) => team !== undefined),
-      distinctUntilChanged(([previousIsV2, previousTeam], [currentIsV2, currentTeam]) =>
-        previousIsV2 === currentIsV2 && previousTeam === currentTeam)
-    ).subscribe(([isV2, team]) => {
-      if (isV2) {
-        this.teamId = team.id;
-      } else {
-        this.teamId = team.guid;
-      }
+      filter((team) => team !== undefined)
```
Comment (msorens) [inline components/automate-ui/src/app/modules/team/team-details/team-details.component.ts]:
```suggestion
      filter(identity)
```

File: components/automate-ui/src/app/modules/team/team-management/team-management.component.html [hunk 6]
```diff
@@ -6,31 +6,22 @@
...
-      </app-create-v1-team-modal>
-
       <app-create-object-modal
         [visible]="createModalVisible"
         [creating]="creatingTeam"
         [conflictErrorEvent]="conflictErrorEvent"
         objectNoun="team"
         [createForm]="createTeamForm"
-        [showProjectsDropdown]="isIAMv2"
+        [showProjectsDropdown]="true"
```
Comment (msorens) [inline components/automate-ui/src/app/modules/team/team-management/team-management.component.html]: Now we could even eliminate this parameter completely if we modify the modal to just check for `objectNoun != 'project'`. That alleviates a bit of the code smell of having 9 parameters(!) to the component.

File: components/automate-ui/src/app/modules/team/team-management/team-management.component.html [hunk 7]
```diff
@@ -52,10 +43,9 @@
                       <span>Name</span>
```
Comment (msorens) [inline components/automate-ui/src/app/modules/team/team-management/team-management.component.html]: Without exception, all other uses do **not** wrap these in `<span>` elements so should now be able to just do:
```
<chef-table-header-cell class="name-column">Name</chef-table-header-cell>
```

(And similarly for L46, L49, and L60.)

File: components/automate-ui/src/app/modules/team/team-details/team-details.component.spec.ts [hunk 8]
```diff
@@ -84,7 +82,7 @@ describe('TeamDetailsComponent v2', () => {
       }
     },
```
Comment (msorens) [inline components/automate-ui/src/app/modules/team/team-details/team-details.component.spec.ts]: Change the test name just above from 'TeamDetailsComponent v2' to 'TeamDetailsComponent'.

File: components/automate-ui/src/app/entities/policies/policy.requests.ts [hunk 9]
```diff
@@ -28,10 +22,6 @@ export class PolicyRequests {
 
   constructor(private http: HttpClient) { }
 
-  public getIamVersion(): Observable<IamVersionResponse> {
-    return this.http.get<IamVersionResponse>(`${env.auth_v2_url}/policy_version`);
```
Comment (msorens) [inline components/automate-ui/src/app/entities/policies/policy.requests.ts]: Should similarly remove `policy_version` calls from components/automate-ui/e2e/admin.e2e-spec.ts

File: components/automate-ui/src/app/components/settings-sidebar/settings-sidebar.component.spec.ts [hunk 10]
```diff
@@ -62,75 +61,69 @@ describe('SettingsSidebarComponent', () => {
...
+        'allOf',
+        elements[i].getAttribute('ng-reflect-all-of'),
+        settingsLandingComponent.routeList[i].allOfCheck);
+    }
   });
 
-  describe('IAM v1', () => {
-    it('shows 8 links', () => {
-      component.isIAMv2$ = observableOf(false);
+  it('shows 11` links', () => {
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/automate-ui/src/app/components/settings-sidebar/settings-sidebar.component.spec.ts]: Always for deleting more code!

File: components/automate-ui/src/app/modules/team/team-management/team-management.component.html (hunk 6 above)
Comment (tylercloke) [reply components/automate-ui/src/app/modules/team/team-management/team-management.component.html]: It's still used in the project create modal.

Comment (msorens) [reply components/automate-ui/src/app/modules/team/team-management/team-management.component.html]: Exactly, please re-read my comment. 😁

Comment (bcmdarroch) [reply components/automate-ui/src/app/modules/team/team-management/team-management.component.html]: ➕ to removing the `showProjectsDropdown` parameter

**Comment by bcmdarroch** [comment]
One thing I forgot to mention: I can take care of dropping the old `auth_url` from [environment.ts](https://github.com/chef/automate/blob/master/components/automate-ui/src/environments/environment.ts#L17) in my PR, since we don't have a `v2` URL for introspection just yet

File: .expeditor/buildkite/cypress.sh [hunk 11]
```diff
@@ -6,21 +6,13 @@ cd /workdir/e2e
 
 data=$(curl --silent "https://a2-${CHANNEL}.cd.chef.co/assets/data.json")
 instances_to_test=$(jq -nr --argjson data "$data" '$data[] | select(.tags | any(. == "e2e")) | .fqdn')
-versions=(v1 v2.1)
```
[the remark this answers is not in the document]
Comment (tylercloke) [reply .expeditor/buildkite/cypress.sh]: https://github.com/chef/automate/issues/2782

**Pull request merged**